pkg/utils/flow/taskfn_test.go

```diff
@@ -138,4 +141,71 @@ var _ = Describe("task functions", func() {
 			Eventually(cancelled).Should(BeClosed())
 		})
 	})
+
+	Describe("ParallelN", func() {
+		It("should run the tasks", func() {
+			var (
+				mut         = sync.Mutex{}
+				m           = map[string]struct{}{}
+				ctx         = context.Background()
+				n           = 2
+				activeTasks int32
+				wg          = sync.WaitGroup{}
+				done        = make(chan struct{})
+				fn          = func(key string) flow.TaskFn {
+					return func(ctx context.Context) error {
+						atomic.AddInt32(&activeTasks, 1)
+						defer atomic.AddInt32(&activeTasks, -1)
+
+						// wait for some time so other tasks can start
+						time.Sleep(time.Millisecond * 100)
```

Comment:
To not rely on timings, you could unlock them in the test after you check that the number of concurrent tasks are active.

For instance:
- Add 3 tasks to ParallelN(2, ...)
- A tasks stops until unblocked from outside, after filling `activeTasks`
- Exptect `activeTasks == 2`
- Unblock all tasks
- Expect `m == len(3)`

WDYT?

Reply:
Thanks for the suggestion @timuthy ! I applied some of the proposed changes, however I still have a single wait in there. I was thinking that I can unblock on the 2nd active task, but then I do not give the opportunity for a 3rd unwanted task to become active. PTAL

Reply:
I pushed [this commit](https://github.com/timuthy/gardener/commit/e2664ba44ea3a7bbcff972dacaefc0c2f2bd3835) to show what I meant. Please let me know what you think.

Reply:
Thanks @timuthy for reviewing and taking the time to refactor the test. I applied your suggestions and I added some clarification comments + a single assertion. PTAL
```golang
Expect(taskIds.Len()).To(Equal(0))
```